spec/services/course/duplication/course_duplication_service_spec.rb

```diff
@@ -410,12 +410,7 @@
 
         it 'duplicates achievement with attached badge' do
           new_achievement_with_badge = new_course.achievements.select(&:badge_url).first
-          new_achievement_id = new_achievement_with_badge.id
-          badge_folder = Rails.root.join('public', 'uploads', 'images', 'course',
-                                         'achievement', new_achievement_id.to_s, 'badge')
-          badge_file = badge_folder.join('minion.png')
-          expect(File.directory?(badge_folder)).to be true
-          expect(File.exist?(badge_file)).to be true
+          expect(File.exist?(File.join(Rails.root, 'public', new_achievement_with_badge.badge.url))).to be true
```

Comment:
better to split this into
```rb
badge_file = ...
expect(File.exist?(badge_file)).to be true
```

1. fewer code changes
2. explicitly declare badge file path

Reply:
done